openmc/universe.py

```diff
@@ -396,6 +401,35 @@ def plot(self, origin=(0., 0., 0.), width=(1., 1.), pixels=(200, 200),
                 height = pixels[0]*px/(params.top - params.bottom)
                 fig.set_size_inches(width, height)
 
+            # add legend showing which colors represent which material
+            # or cell if that was requested
+            if legend:
+                if plot.colors is None:
+                    raise Exception("Must set plot color dictionary if you "
+                                    "would like to have an automatic legend.")
```

Comment:
I would change this to `ValueError` and be a little more explicit about what is expected (Must pass 'colors' argument...)